### PR TITLE
CI / Guthub actions v2

### PR DESCRIPTION
**Pull request by Frank1234**
## Why is this important?

We had a two-headed CI setup, partly fastlane, partly github actions. It increased complexity * 2. Also, signing key usage was not included, we've added it here.

## In this ticket
- included workflows for building the apps, inc. a composit workflow for shared logic.
- any commit on any branch triggers a `devDebug` apk and a `prodDebug` apk build
- any PR triggers signed release builds (`prodRelease` bundle and `prodRelease` apk)
- added lots of documentation in README on using CI and setting the correct secrets etc. https://github.com/Q42/Template.Android/blob/782d6852bf3c143ca41c74ea3c416b97d7ea016c/README.MD#ci--automated-builds
- upload to Firebase Distribution on merges to develop
- run unittests on all commits

Example actions that ran because of this PR:
- debug: https://github.com/Q42/Template.Android/actions/runs/10176805470
- release: https://github.com/Q42/Template.Android/actions/runs/10176805837

### CODE REVIEW

**Review by Techwolf12 — DISMISSED** [review]

File: .github/workflows/_prepare-all.yml
Comment (ninovanhooff) [inline .github/workflows/_prepare-all.yml:8]: Price of Mac runners is 10x that of linux: https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#minute-multipliers

What is your reason for choosing mac? Did you try running on linux? If not, I would say it is at least worth the effort to do a test run.

edit: it tried and it failed. For an unexpected reason: it couldn't find the requested version of firebase
Up to you whether you feel this is worth looking into.

File: README.MD
Comment (ninovanhooff) [inline README.MD:35]: Having the wishlist in the README.md is discovererable, but also very rigid, since edits require approval. 

I would prefer these to be converted to Issues with the label `improvement` or `feature` applied.

Also, I feel slightly anxious when picking up one of these, for fear that someone else is working on it. I would prefer to claim an issue by self-assignment

File: gradle/libs.versions.toml
Comment (ninovanhooff) [inline gradle/libs.versions.toml:15]: Please review deprecations warnings regarding old github action versions: https://github.com/Q42/Template.Android/actions/runs/10307994442

File: README.MD
Comment (ninovanhooff) [inline README.MD:82]: After merging this PR, please trigger ask dependabot build by choosing any open dependabot PR and reacting with `@dependabot recreate` 

If that fails, the changes you made in the secrets need to be reflected into the dependabot secrets; and that should be documented here

Comment (ninovanhooff) [reply README.MD:82]: See https://github.com/Q42/Template.Android/settings/secrets/dependabot

File: README.MD
Comment (ninovanhooff) [inline README.MD:129]: Well documented, thanks! 👍

File: README.MD
Comment (ninovanhooff) [inline README.MD:116]: https://www.notion.so/q42/GitHub-Actions-CI-op-onze-self-hosted-runner-Mac-Mini-7f7d2e6312444b98be521394223db6e2?pvs=4#99626170e0e24fc1ae48d30f88d96b59

Unsure whether this could work outside q42 org

File: app/build.gradle
Comment (ninovanhooff) [inline app/build.gradle:16]: why not 42 ;-)

**Review by ninovanhooff — DISMISSED** [review]
This all looks good, thanks!

None of my comments are blocking, so I'll approve. 

But please do consider them :-)

File: .github/workflows/_prepare-all.yml
Comment (ninovanhooff) [reply .github/workflows/_prepare-all.yml:8]: Could be related to java version incompatibility somehow, even though java 21 is mentioned: https://github.com/firebase/firebase-android-sdk/issues/6123

Comment (Frank1234) [reply .github/workflows/_prepare-all.yml:8]: Because we use our own mac-mini on our projects, that is free. Switching from mac -> mac-mini is no effort.

I added documentation on how to use our own mac-mini. In the "Using this template" step, "You probably want to enable the self-hosted runner". I'll update it a bit to be even more clear.

File: README.MD
Comment (Frank1234) [reply README.MD:116]: Probably not, you can then let the customer pay :)

File: gradle/libs.versions.toml
Comment (Frank1234) [reply gradle/libs.versions.toml:15]: thanks, I will

Comment (Frank1234) [reply gradle/libs.versions.toml:15]: done! great improvement

File: README.MD
Comment (Frank1234) [reply README.MD:36]: great idea! I've added this wish to the wishlist :)

**Pull request merged**
